ccd-config-generator/src/main/java/uk/gov/hmcts/ccd/sdk/Main.java

```diff
@@ -1,32 +1,38 @@
 package uk.gov.hmcts.ccd.sdk;
 
 import java.io.File;
-import java.util.Set;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.SneakyThrows;
 import org.reflections.Reflections;
-import org.reflections.util.ClasspathHelper;
-import org.reflections.util.ConfigurationBuilder;
-import org.springframework.boot.SpringApplication;
-import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.ConfigurableApplicationContext;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
 
 class Main {
 
   public static void main(String[] args) {
-    Reflections reflections = new Reflections(new ConfigurationBuilder()
-        .setUrls(ClasspathHelper.forPackage(args[1]))
-        .setExpandSuperTypes(false));
+    File outputDir = new File(args[0]);
 
-    Set<Class<?>> types =
-        reflections.getTypesAnnotatedWith(SpringBootApplication.class);
-    if (types.size() != 1) {
-      throw new RuntimeException("Expected a single SpringBootApplication but found "
-          + types.size());
-    }
-    try (ConfigurableApplicationContext context =
-        SpringApplication.run(types.iterator().next(), args)) {
+    Reflections reflections = new Reflections(args[1]);
+    List<CCDConfig> configs = reflections
+        .getSubTypesOf(CCDConfig.class)
+        .stream()
+        .map(configClass -> getCcdConfig(configClass))
+        .flatMap(Optional::stream)
+        .collect(Collectors.toList());
+
+    ConfigGenerator generator = new ConfigGenerator(configs);
+    generator.resolveConfig(outputDir);
+    // Required on Gradle 4.X or build task hangs.
+    System.exit(0);
+  }
 
-      File outputDir = new File(args[0]);
-      context.getBean(ConfigGenerator.class).resolveConfig(outputDir);
+  @SneakyThrows
+  private static Optional<CCDConfig> getCcdConfig(Class<? extends CCDConfig> configClass) {
+    try {
+      return Optional.of(configClass.getDeclaredConstructor().newInstance());
+    } catch (NoSuchMethodException e) {
```

Comment:
Does this require every config to have a default constructor? could use objenisys to get an instance if we still have that dependency

Reply:
It doesn't. I tested in NFD and classes like: ```
@Component
public class CreateDraft implements CCDConfig<CaseData, State, UserRole> {

    public static final String CREATE_DRAFT = "create-draft";

    @Override
    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {

        configBuilder
            .event(CREATE_DRAFT)
            .initialState(Draft)
            .name("Create draft case")
            .description("Apply for a divorce or dissolution")
            .grant(CREATE_READ_UPDATE, CITIZEN)
            .displayOrder(1)
            .retries(120, 120);
    }
}
```

Are processed correctly. I had to add the catch because the inner class in one of the tests was causing an issue. I'm not sure why, perhaps the constructor of the inner class is private (only accessible by the outer class?).